SConstruct

```diff
@@ -748,3 +752,16 @@ if "env" in locals():
     # TODO: replace this with `env.Dump(format="json")`
     # once we start requiring SCons 4.0 as min version.
     methods.dump(env)
+
+
+def print_elapsed_time():
+    elapsed_time_sec = time.time() - time_at_start
+    if elapsed_time_sec < 60:
+        print(f"[Time elapsed: {elapsed_time_sec:.1f} seconds.]")
+    else:
+        print(
+            f"[Time elapsed: {elapsed_time_sec // 60} minute{'' if elapsed_time_sec < 120 else 's'} and {(elapsed_time_sec % 60):.1f} seconds.]"
+        )
```

Comment:
Wouldn't it be simpler to just show it as `HH:MM:SS` and avoid having to handle pluralization?

The current code doesn't handle plurals properly for "1 seconds" and for 0 hours or 0 seconds (which should also be plural in English IINM), and I'm not sure making it more complex to handle plurals would be beneficial when a numeral form is probably easier to parse in successive builds (even programmatically).

Reply:
I think having a decimal number of seconds needs the pluralised form of seconds, but I do agree that HH:MM:SS would be a lot simpler and easier to parse.

Reply:
I wrote it like that to display the elapsed time it in a appealing/human readable way for most cases, but I think your suggestion for a standard HH:MM:SS format is actually better.

Reply:
Would we omit the milliseconds and only show whole number values for seconds?

Reply:
As Calinou mentioned earlier, for something like build time showing only whole seconds is enaugh.

Reply:
I'm fine with both, if we want milliseconds it can be `HH:MM:SS.mmm`. Build times even for an empty rebuild are typically at least 6 s, but if we want to microoptimize some aspects having milliseconds might be useful.

Reply:
Do we need a custom HH:MM:SS formatting system, as this current method would lapse at 24 hours, though it likely isn't needed in the scope of this implementation.

Reply:
I'd say keep it simple, improve if/when there's a need :)

Reply:
Changed format to `HH:MM:SS.mmm`. Since time's strftime does not support `%f` and I couldn't find a similar way to do it with datetime and timedeltas I had to calculate it manually (but it's nice and short, so it should be ok).